staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth.go

```diff
@@ -125,13 +132,25 @@ func GetServicePrincipalToken(config *AzureAuthConfig, env *azure.Environment) (
 	return nil, ErrorNoAuth
 }
 
-// ParseAzureEnvironment returns azure environment by name
-func ParseAzureEnvironment(cloudName string) (*azure.Environment, error) {
+// ParseAzureEnvironment returns the azure environment.
+// If 'cloudFQDN' is set, environment is computed by quering the cloud's resource manager endpoint.
+// Otherwise, a pre-defined Environment is looked up by name.
+func ParseAzureEnvironment(cloudName, cloudFQDN, identitySystem string) (*azure.Environment, error) {
 	var env azure.Environment
 	var err error
-	if cloudName == "" {
+	if cloudFQDN != "" {
```

Comment:
@vpatelsj, does this work for you as well?

Reply:
hrmm, not quite, I kinda need to be able to completely override the resourceManagerEndpoint. This will force me to have a subdomain `managment` in my proxy server.

Reply:
Not a problem, can do that later today/tomorrow.
@andyzhangx, @feiskyer, please comment if you see a problem